src/test/AludelFactoryIntegration.sol

```diff
@@ -115,269 +125,111 @@ contract AludelFactoryIntegrationTest is DSTest {
-            )
-        );
-        assertEq(aludel.getBonusTokenSetLength(), 2);
-
-        assertEq(factory.programs(address(aludel)).template, address(template));
-        assertEq(factory.programs(address(aludel)).name, "name");
-        assertEq(factory.programs(address(aludel)).startTime, block.timestamp);
-
-        IAludel.AludelData memory aludelData = aludel.getAludelData();
-
-        MockERC20(aludelData.rewardToken).mint(owner, 1 ether);
-
-        cheats.startPrank(owner);
-        MockERC20(aludelData.rewardToken).approve(address(aludel), 1 ether);
-        aludel.fund(1 ether, 1 days);
-        aludel.registerVaultFactory(address(template));
-        cheats.stopPrank();
-
-        aludelData = aludel.getAludelData();
-        IAludel.RewardSchedule[] memory rewardSchedules = aludelData
-            .rewardSchedules;
-        assertEq(rewardSchedules[0].shares, 0.99 ether * BASE_SHARES_PER_WEI);
-    }
-
-    function test_template_initialization() public {
-        AludelV2 template = new AludelV2();
-        template.initializeLock();
-        factory.addTemplate(address(template), "bleep", false);
+        Utils.fundMockToken(address(crucible), stakingToken, STAKE_AMOUNT);
     }
 
-    function testFail_template_double_initialization() public {
-        AludelV2 template = new AludelV2();
-        template.initializeLock();
-        cheats.expectRevert(new bytes(0));
-        template.initializeLock();
-    }
 
-    function test_stake() public {
-        bytes memory permission = getPermission(
-            PRIVATE_KEY,
-            "Lock",
+    function test_GIVEN_default_program_WHEN_user_stakes_THEN_succeeds() public {
+        Utils.stake(
+            user,
             crucible,
-            address(aludel),
-            address(stakingToken),
-            1 ether
+            aludel,
+            stakingToken,
+            STAKE_AMOUNT
         );
-        cheats.prank(owner);
-        aludel.stake(crucible, 1 ether, permission);
     }
 
-    function test_unstake() public {
-        IAludel.VaultData memory vault = aludel.getVaultData(crucible);
+
+    function test_GIVEN_default_program_AND_staked_user_WHEN_user_unstakes_THEN_succeeds() public {
+
+        IAludel.VaultData memory vault = aludel.getVaultData(address(crucible));
         assertEq(vault.totalStake, 0);
         assertEq(vault.stakes.length, 0);
 
-        bytes memory lockPermission = getPermission(
-            PRIVATE_KEY,
-            "Lock",
+        Utils.stake(
+            user,
             crucible,
-            address(aludel),
-            address(stakingToken),
-            1 ether
+            aludel,
+            stakingToken,
+            STAKE_AMOUNT
         );
-        cheats.prank(owner);
-        aludel.stake(crucible, 1 ether, lockPermission);
 
-        vault = aludel.getVaultData(crucible);
-        assertEq(vault.totalStake, 1 ether);
+        vault = aludel.getVaultData(address(crucible));
```

Comment:
I think all of the assertions that are a consequence of _staking_ should go in the previous testcase, and this one should assume staking works okay and focus on asserting what happens _after_ unstaking